pandas/_libs/tslib.pyx

```diff
@@ -595,6 +645,42 @@ cpdef array_to_datetime(
                         iresult[i] = NPY_NAT
                         continue
 
+                    if (
+                        format is not None
+                        and (
+                            not require_iso8601
+                            or (
+                                require_iso8601 and format == "%Y%m%d" and len(val) != 8
+                            )
+                        )
+                        and val not in ("today", "now")
+                    ):
+                        try:
+                            _iresult, _tzinfo = strptime(
+                                val, format, exact, format_regex, locale_time, dts
+                            )
+                        except (ValueError, OverflowError):
+                            if is_coerce:
+                                iresult[i] = NPY_NAT
+                                continue
+                            elif is_raise:
+                                raise
+                            return values, tz_out
+                        value = tz_localize_to_utc_single(_iresult, _tzinfo)
+                        if _tzinfo is not None:
+                            found_tz = True
+                            tz_out = convert_timezone(
+                                _tzinfo,
+                                tz_out,
+                                found_naive,
+                                found_tz,
+                                utc_convert,
+                            )
+                        else:
+                            found_naive = True
```

Comment:
the thing with `_tzinfo` here is that it might not be a fixed offset. E.g. it could be `pytz.timezone('US/Pacific')`. So, I don't think we can take the same approach as is done below in the ISO8601 fastpath of just saving the offset minutes from UTC to `out_tzoffset`